provider/management/commands/delete_expired_grant_tokens.py

```diff
@@ -0,0 +1,55 @@
+"""
+Command to delete expired OAuth2 grant tokens from the oauth2_grant table.
+
+Accepts a parameter to determine how many days back to consider a token expired.
+"""
+
+import logging
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+
+from provider.oauth2.models import Grant
+
+log = logging.getLogger(__name__)
+
+class Command(BaseCommand):
+    """
+    Example usage: ./manage.py lms --settings=devstack delete_expired_grant_tokens.py --days_back 2
+    """
+    help = 'Deletes OAuth2 grant tokens expired for longer than days_back days.'
+
+    # Default value for days_back.
+    DEFAULT_DAYS_BACK = 1
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--days_back',
+            nargs='+',
+            default=self.DEFAULT_DAYS_BACK,
+            type=int,
+            help='Delete tokens expired for longer than this many days.'
+        )
+
+    def handle(self, *args, **options):
+        """
+        Deletes oauth2_grant tokens expired for days_back days, chunking the deletes to avoid long table/row locks.
+        """
+        CHUNK_SIZE = 10000
+        days_back = options.get('days_back', self.DEFAULT_DAYS_BACK)
+        if days_back < 0:
+            raise CommandError('Only non-negative days back are allowed ({}).'.format(days_back))
+        days_back *= -1
+        where_clause = 'WHERE expires < TIMESTAMPADD(DAY, {}, NOW())'.format(days_back)
+        total_expired_tokens = Grant.objects.raw(
+            'SELECT COUNT(*) FROM oauth2_grant {}'.format(where_clause)
```

Comment:
The "expires" column doesn't seem to be indexed in a way where it will help this. My experience on the read replica was that these queries took several seconds. Obviously this will get better after the first time this script is run, but I'd avoid doing the select and just opt for the delete limited by CHUNK_SIZE to cut down on the database pain.